src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -415,42 +415,34 @@ private void WriteStatusCodeSwitch(CodeWriter writer, ResponseBody? responseBody
                 using (responseBody != null ? writer.Scope() : default)
                 {
                     string valueVariable = "value";
-
+                    string messageVariable = "message";
+                    string responseVariable = $"{messageVariable}.Response";
                     if (responseBody is ObjectResponseBody objectResponseBody)
                     {
-                        const string document = "document";
                         switch (objectResponseBody.Serialization)
                         {
                             case JsonSerialization jsonSerialization:
-                                writer.Append($"using var {document:D} = ");
-                                if (async)
-                                {
-                                    writer.Line($"await {typeof(JsonDocument)}.ParseAsync(message.Response.ContentStream, default, cancellationToken).ConfigureAwait(false);");
-                                }
-                                else
-                                {
-                                    writer.Line($"{typeof(JsonDocument)}.Parse(message.Response.ContentStream);");
-                                }
-
-                                writer.ToDeserializeCall(
-                                    jsonSerialization,
-                                    w => w.Append($"document.RootElement"),
-                                    ref valueVariable
-                                );
+                                writer.WriteMethodDeserialization(jsonSerialization, async, ref valueVariable, responseVariable);
                                 break;
                             case XmlElementSerialization xmlSerialization:
-                                writer.Line($"var {document:D} = {typeof(XDocument)}.Load(message.Response.ContentStream, LoadOptions.PreserveWhitespace);");
-                                writer.ToDeserializeCall(
-                                    xmlSerialization,
-                                    w => w.Append($"document"),
-                                    ref valueVariable
-                                );
+                                writer.WriteMethodDeserialization(xmlSerialization, ref valueVariable, responseVariable);
                                 break;
                         }
                     }
                     else if (responseBody is StreamResponseBody _)
                     {
-                        writer.Line($"var {valueVariable:D} = message.ExtractResponseContent();");
+                        writer.Line($"var {valueVariable:D} = {messageVariable:D}.ExtractResponseContent();");
+                        using (writer.Scope($"if ({valueVariable:D} == null)"))
```

Comment:
We don't check that content isn't null in other branches why do it here?

Reply:
The other serializers create non-null Value variables (because of their deserialization mechanics). Here, we don't get to do that. The other deserializers will error, however, if it is null because it cannot parse the content. So, there's not a need to throw a different error in that situation.

Reply:
I'm not sure I'm a fan of how the Exception would look. This might result in an exception with  "200 success" status code in a message. Let's create an extension method for HttpPipelineMessage that checks stream for null and throws some reasonable exception.

Reply:
We already have failures that result in 200 success, as we default to this same throw at the bottom of this method (status code switch's default). While I agree with you, I think it makes more sense to look at our throws that we write as a whole and clean them up in a single PR. I love refactoring, but we need to be complete when doing it (not just looking at a specific situation at a time).

Reply:
Let's skip this check here then and add them to all places all at once.

Reply:
Okay. Just FYI, that means it will technically be returning `Response<Stream?>` instead of `Response<Stream>`. Since we don't have nullability for those projects, they don't care, though.